src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -767,6 +767,8 @@ private void ApplyAdditionalProperties<TSchemaType>(TSchemaType schema, Type typ
             {
                 var genericTypeArguments = extensionDataProperty.GenericArguments;
                 var extensionDataPropertyType = genericTypeArguments.Length == 2 ? genericTypeArguments[1] : typeof(object).ToContextualType();
+                if (extensionDataPropertyType.TypeInfo.FullName.Equals("System.Text.Json.JsonElement"))
+                    extensionDataPropertyType = typeof(object).ToContextualType();
```

Comment:
I think its better if we handle this here:
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/Generation/DefaultReflectionService.cs#L177

This would ensure that its always handled as object not only in the case of extension data.. also is element an object ({}) or any (also primitive value allowed).

Reply:
Moved the check to DefaultReflectionService.